homeassistant/components/minecraft_server/__init__.py

```diff
@@ -0,0 +1,335 @@
+        await self.async_check_connection()
+        server_online = self.online
+
+        # Inform user once about connection state changes if necessary.
+        if (server_online_old is True) and (server_online is False):
+            _LOGGER.warning("Connection to server lost")
+        elif (server_online_old is False) and (server_online is True):
+            _LOGGER.info("Connection to server (re-)established")
+
+        # Try to update the server data if server is online.
+        if server_online:
+            try:
+                await self._async_status_request()
+            except IOError as error:
+                _LOGGER.debug(
+                    "Error occurred while trying to update the server data: %s", error
+                )
+        else:
+            # Set all properties except description and version information to
+            # unknown until server connection is established again.
+            self._players_online = None
+            self._players_max = None
+            self._players_list = None
+            self._latency = None
+
+        # Notify sensors about new data.
+        async_dispatcher_send(self._hass, self._signal_name)
+
+        # Periodically update status.
+        self._stop_periodic_update = async_track_point_in_utc_time(
+            self._hass,
+            self.async_update,
+            utcnow() + timedelta(seconds=self._scan_interval),
+        )
+
+    async def _async_status_request(self) -> None:
+        """Request server status and update properties."""
+        try:
+            status_response = await self._hass.async_add_executor_job(
+                self._mcstatus.status, self._RETRIES_STATUS
+            )
+        except IOError:
+            _LOGGER.debug("Error while requesting server status: IOError")
+            raise IOError
+        else:
+            self._description = status_response.description["text"]
+            self._version = status_response.version.name
+            self._protocol_version = status_response.version.protocol
+            self._players_online = status_response.players.online
+            self._players_max = status_response.players.max
+            self._latency = status_response.latency
+            self._players_list = []
+            if status_response.players.sample is not None:
+                for player in status_response.players.sample:
+                    self._players_list.append(player.name)
+
+
+class MinecraftServerEntity(Entity):
+    """Representation of a Minecraft Server base entity."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        server: MinecraftServer,
+        type_name: str,
+        icon: str,
+        device_class: str,
+    ) -> None:
+        """Initialize base entity."""
+        self._server = server
+        self._hass = hass
+        self._name = f"{server.name} {type_name}"
+        self._icon = icon
+        self._unique_id = f"{self._server.unique_id}-{type_name}"
```

Comment:
IP address is not allowed to be used for entity unique_id. IP addresses may change.

Is the mine craft server host never an ip address?

Reply:
https://developers.home-assistant.io/docs/en/entity_registry_index.html#unique-id-requirements

Reply:
Yes, it can be an IP address, too. Hmm, I didn't think about the possibility that an IP address can change, good point. 
Is there a way to get the corresponding MAC address from within HA? Or do I have to ask the user during config flow?

Reply:
It's ok to use the `getmac` library with its limitations.

Reply:
I will have a look into this. Thanks for the hint.